src/View/View.php

```diff
@@ -554,6 +554,11 @@ public function render($view = null, $layout = null)
             return;
         }
 
+        if ($layout !== null) {
+            $defaultLayout = $this->layout;
+            $this->layout = $layout;
+        }
```

Comment:
> Then when rendering the layout we can use $this->layout instead of the local variable.

That's what is being done, check line 570. 

> Why do you need to restore the layout after rendering?

I did it just to keep previous behavior where layout property remained unmodified after rendering finished.